xcdat/regridder/regrid2.py

```diff
@@ -486,21 +402,46 @@ def _pertub(value: xr.DataArray) -> xr.DataArray:
 
     Parameters
     ----------
-    value : xr.DataArray
+    value : np.ndarray
         Value to pertub.
 
     Returns
     -------
-    xr.DataArray
+    np.ndarray
         Value that's been pertubed.
     """
     if value >= 0.0:
         offset = np.ceil(value + 0.000001)
     else:
         offset = np.floor(value - 0.000001) + 1.0
 
-    return xr.DataArray(offset)
+    return offset
 
 
 # vectorize version of pertub
 _vpertub = np.vectorize(_pertub)
+
+
+def _get_dimension(input_data_var, cf_axis_name):
+    name = input_data_var.cf.axes[cf_axis_name]
+
+    if isinstance(name, list):
+        name = name[0]
```

Comment:
Unless the intent here is to only interpret the `axis` CF attribute?

Instead of using `cf_xarray` directly, I think you can use [xc.get_dim_keys()](https://xcdat.readthedocs.io/en/stable/_modules/xcdat/axis.html#get_dim_keys) which can also interpret the `standard_name` attribute or use the xCDAT fall-back table of generally accepted axis names.

Reply:
I could use `get_dim_keys` but there will be a trade off of robustness for performance. If we accept this I'm fine making the change.

Reply:
I see. Is the performance hit significant using `get_dim_keys()`. If so, I think it is fine to only interpret the `axis` attribute for performance.

@lee1043 any thoughts?

Reply:
Try passing data variable directly to `get_dim_keys()`

Reply:
Looks like using `get_dim_keys` works just fine and no decrease in performance, actually a small increase.